### PR TITLE
feat: add support for py312

### LINKED ISSUE #1355: Enable python 3.12 in Github Action runner

### PR DESCRIPTION
**Pull request by chebbyChefNEQ**
closes #1355

GHA released py312 two weeks ago

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ jobs:
     timeout-minutes: 45
     strategy:
       matrix:
-        python-minor-version: [ "8", "9", "10", "11" ]
+        python-minor-version: [ "8", "9", "10", "11", "12" ]
```
Comment (wjones127) [inline .github/workflows/python.yml:75]: @changhiskhan Do you think we get value out of running the entire matrix? Or would it be just as good to run the min and max versions?

Comment (eddyxu) [reply .github/workflows/python.yml:75]: +1

Comment (wjones127) [reply .github/workflows/python.yml:75]: BTW Pandas and Numpy don't support Python 3.8 anymore. Should we?

Comment (eddyxu) [reply .github/workflows/python.yml:75]: numpy 1.22/23 still support py3.8 tho

https://endoflife.date/numpy#:~:text=All%20minor%20versions%20of%20Python,never%20on%20a%20patch%20release.&text=NumPy%20stopped%20supporting%20Python%203.7,1.21)%20still%20support%20Python%203.7.

Comment (wjones127) [reply .github/workflows/python.yml:75]: I see that's fair.

**Pull request closed without merging**